### ISSUE TITLE
Query: Different behavior of toggling flagged and other toggles in index view?

### ISSUE THREAD

**Issue by hgvhgv** [wangp/bower#110]
In the index view, what is the thinking behind refusing to toggle the flagged tag on threads with more than one message? `U`, `a`, `d`,`u`,`$` all work as described, tagging/toggling all messages in the thread. But `F` refuses whenever there is more than one message. It would be good to specify this in the readme too.

**Comment by wangp** [comment]
I use the `flagged` tag to indicate a message as important or something to come back to later. I think there's no further justification than it didn't make sense to me to mark every message in a thread as important.

**Comment by hgvhgv** [comment]
Right, that makes sense (and is how I use it too). But I think the readme should clarify the difference between different toggle actions in the index view. Should I make a PR or do you just want to manage it yourself?

On a related note: when setting or toggling a tag in index view, it might make sense to apply the action only to messages in the thread that match the current search (at least for certain tags). For me at least, this would solve the flag toggle for new messages, which arrive in the default inbox attached to threads that aren't there; clicking `F` would then just apply the flagged tag to the new unread message in the thread. I have a few other tags I use in a similar manner. I'll try to dummy this up in my branch, but if it's of interest I'll submit a PR.

**Comment by wangp** [comment]
Sure, send a PR if you like.

On the other suggestion, I'm afraid it might be too confusing or non-obvious. I usually don't think about which messages matched in a thread.

**Comment by haasn** [comment]
> I use the `flagged` tag to indicate a message as important or something to come back to later. I think there's no further justification than it didn't make sense to me to mark every message in a thread as important.

To me it makes perfect sense to mark a *thread* as important, without carrying about any particular message in this thread. The equivalent of keeping a tab of an issue open for later, I suppose.

**Comment by wangp** [comment]
The notmuch model doesn't support tags on threads. However, we could flag the first message in the thread instead.

**Comment by hgvhgv** [comment]
I took tagging a thread to mean just tagging all messages in the thread (so like how the non-flagged toggling/tagging actions in the index view currently function). Along those lines, it wasn't clear to me why`F` should act differently than `d` (the former not tagging all msgs while the latter does), but clarifying the documentation seems enough to me. I do not understand tagging the first message in a thread rather than all messages (especially if the first message didn't itself match the search query) or just the msgs that match the query. But in the end, I think people use notmuch tags pretty idiosyncratically and it's fine to follow your preferences for the project.
